Description: If EDITOR or VISUAL is not set, fall back to /usr/bin/editor, which is set by update-alternatives.
Author: Manos Pitsidianakis <manos@pitsidianak.is>
Last-Update: 2024-11-19
Index: meli/meli/src/subcommands.rs
===================================================================
--- meli.orig/meli/src/subcommands.rs
+++ meli/meli/src/subcommands.rs
@@ -56,9 +56,7 @@
 pub fn edit_config() -> Result<()> {
     let editor = std::env::var("EDITOR")
         .or_else(|_| std::env::var("VISUAL"))
-        .map_err(|err| {
-            format!("Could not find any value in environment variables EDITOR and VISUAL. {err}")
-        })?;
+        .unwrap_or_else(|_| "/usr/bin/editor".into());
     let config_path = conf::get_config_file()?;
 
     let mut cmd = Command::new(editor);
